crates/core/src/db/datastore/locking_tx_datastore/state_view.rs

```diff
@@ -318,60 +358,124 @@ impl<'a> Iterator for IndexSeekIterMutTxId<'a> {
             //
             // As a result, in MVCC, this branch will need to check if the `row_ref`
             // also exists in the `tx_state.insert_tables` and ensure it is yielded only once.
-            .and_then(|i| i.find(|row_ref| !self.tx_state.is_deleted(self.table_id, row_ref.pointer())))
+            .and_then(|i| i.find(|row_ref| !self.del_table.contains(&row_ref.pointer())))
         {
             // TODO(metrics): This doesn't actually fetch a row.
             // Move this counter to `RowRef::read_row`.
-            self.num_committed_rows_fetched += 1;
+            // self.num_committed_rows_fetched += 1;
             return Some(row_ref);
```

Comment:
I intentionally keep this as a reminder, I have no context to say why it is there or how to change it.